grakn-core/src/main/java/ai/grakn/GraknConfigKey.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Grakn - A Distributed Semantic Database
+ * Copyright (C) 2016  Grakn Labs Limited
+ *
+ * Grakn is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Grakn is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Grakn. If not, see <http://www.gnu.org/licenses/gpl.txt>.
+ *
+ */
+
+package ai.grakn;
+
+import ai.grakn.util.ErrorMessage;
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableList;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * Class for keys of properties in the file {@code grakn.properties}.
+ *
+ * @param <T> the type of the values of the key
+ * @author Felix Chapman
+ */
+@AutoValue
+public abstract class GraknConfigKey<T> {
```

Comment:
I am not going mad am I? Using Autovalue you are somehow initialising an abstract class??

Reply:
That's how autovalue works 👍 
It generates a concrete implementation of the class, called `AutoValue_GraknConfigKey`